iota-execution/latest/iota-adapter/src/programmable_transactions/execution.rs

```diff
@@ -856,6 +886,136 @@ mod checked {
+                // validated by the iota-verifier during package verification (in
+                // `publish_and_verify_modules`).
+                let runtime_module_metadata: RuntimeModuleMetadata =
+                    bcs::from_bytes::<RuntimeModuleMetadataWrapper>(&md.value)
+                        .map_err(|_| {
+                            ExecutionError::from_kind(
+                                ExecutionErrorKind::VMVerificationOrDeserializationError,
+                            )
+                        })?
+                        .try_into()
+                        .map_err(|_| {
+                            ExecutionError::from_kind(
+                                ExecutionErrorKind::VMVerificationOrDeserializationError,
+                            )
+                        })?;
+
+                // PackageMetadataV1 specific:
+                // - Process functions for each module in order to create function metadata:
+                //    - Authenticator attributes, if present, are extracted to create
+                //      AuthenticatorInfoMetadata to insert into the PackageMetadata
+                let mut module_metadata_map = HashMap::new();
+                for (fn_name, fn_attributes) in runtime_module_metadata.fun_attributes_iter() {
+                    // Check attributes
+                    for attribute in fn_attributes {
+                        match attribute {
+                            IotaAttribute::Authenticator(attribute) if attribute.version == 1 => {
+                                module_metadata_map.insert(
+                                    fn_name.to_string(),
+                                    TypeName::from(&get_authenticator_first_param_type_tag(
+                                        module, &fn_name,
+                                    )?),
+                                );
+                            }
+                            _ => { /* Other attributes are ignored for PackageMetadataV1 */ }
+                        }
+                    }
+                }
+                // Fill the package metadata with a module handle (and its related function
+                // metadata) only if there is at least one function with
+                // relevant metadata
+                if !module_metadata_map.is_empty() {
+                    modules_metadata_map.insert(module.name().to_string(), module_metadata_map);
+                }
+                // End of PackageMetadataV1 specific
+            }
+        }
+
+        // Only publish package metadata if there is at least one module with
+        // relevant metadata
+        if !modules_metadata_map.is_empty() {
+            // Create the package metadata "special" object UID
+            let metadata_uid = context.package_derived_metadata_id(storage_id)?;
+            // Create the package metadata object content
+            let metadata_wrapper = PackageMetadata::new_v1(
```

Comment:
```suggestion
            let metadata = PackageMetadata::new_v1(
```
Not so important, of course, you can ignore it if you wish. We have a different metadata wrapper struct, so it would probably be a bit clearer.

Reply:
[40b2281](https://github.com/iotaledger/iota/pull/9296/commits/40b22810f79420530d390d12c9a21d251a774535)